src/viadot/orchestration/prefect/utils.py

```diff
@@ -28,6 +28,394 @@
 with contextlib.suppress(ModuleNotFoundError):
     from prefect_azure import AzureKeyVaultSecretReference
 
+from typing import Dict, List, Union
+import pendulum
+import re
+
+
+class DynamicDateHandler:
+    def __init__(
+        self,
+        dynamic_date_symbols: List[str] = ["<<", ">>"],
+        dynamic_date_format: str = "%Y%m%d",
+        dynamic_date_timezone: str = "Europe/Warsaw",
+    ):
+        """
+        This class handles pendulum DateTimes or time-related patterns within a provided text, replacing dynamic date marker with actual dates.
+        The supported patterns include:
+        - "today"
+        - "yesterday"
+        - "this month"
+        - "last month"
+        - "this year"
+        - "last year"
+        - "now time"
+        - "last day previous month"
+        - "last day of month year": e.g., "last day of February 2020"
+        - "X years ago": e.g., "3 years ago" , refers to only the year of the day X years ago
+        - "X years/months/days ago full date":  e.g., "3 years ago full date", refers to a given date X units ago in dynamic_date_format
+        - "last X years/months/days": e.g., "last 10 months", refers to a data range of the months in 'YMM' format
+        - "Y years from X": e.g., "10 years from 2020", refers to a data range of the year numbers from a specified year
+        - "first X days from X": e.g., "first 10 days of January 2020", returns a data range of days from a given month
+
+        Args:
+            dynamic_date_symbols (List[str], optional): The symbols that mark the start and the end of a dynamic date pattern in a text. Defaults to ["<<", ">>"].
+            dynamic_date_format (str, optional): A date and time format string defining the text representation of date. Defaults to "%Y%m%d".
+            dynamic_date_timezone (str, optional): A string that sets the default timezone used by all datetime functions. Defaults to "Europe/Warsaw".
+        """
```

Comment:
Add `# noqa: D205` at the end. There is an issue with precommit saying that `1 blank line required between summary line and description`
```
""" text """ # noqa: D205
```